hs_core/tasks.py

```diff
@@ -298,24 +302,45 @@ def delete_zip(zip_path):
 
 @shared_task
 def create_temp_zip(resource_id, input_path, output_path, sf_aggregation):
+    """ Create temporary zip file from input_path and store in output_path
+    :param input_path: full irods path of input starting with federation path
+    :param output_path: full irods path of output starting with federation path
+    :param sf_aggregation: if True, include logical metadata files
+    """
     from hs_core.hydroshare.utils import get_resource_by_shortkey
-    if sf_aggregation:
-        pass
     res = get_resource_by_shortkey(resource_id)
-    full_input_path = '{root_path}/{path}'.format(root_path=res.root_path, path=input_path)
-    istorage = IrodsStorage()
+    istorage = res.get_irods_storage()  # invoke federated storage if necessary

```

Comment:
So it was intentional to not invoke federated storage and only use local storage for temporary zips.  I didn't see a reason to store a temporary zipped folder in federated locations.  This was my first task in hydroshare... so I'm sure I didn't and don't have the same grasp of the system as you do, but do we really need to change this?  @hyi reviewed it as well, so maybe she can comment.

Reply:
@sblack-usu Now that you brought this up, I remember we had some discussion as part of your PR review regarding whether we can store this temporary zipped folder in federated location or not. Both would work, and since two iRODS servers are federated, there may not be enough performance impact one way or the other. Those zipped temp folder files are temporary and will be deleted after being downloaded, so it does not have quota impact. @alvacouch may comment on whether there is actually performance benefit to store the temporarily generated zipped folder in a centralized temp collection in the data zone regardless where the resource is stored, federated zone or not. I don't have strong feelings on this since they both should work. The only potential implication I can think of is that for a large folder zip file, disk space may be a consideration factor to store it where the resource is stored, rather than the centralized temp collection in data zone. However, since the temp folder zip file will be cleaned up after being downloaded anyway, that is not a convincing reason to influence our decision. I guess I am trying to say I am good either way and don't see much implication one way or the other.

Reply:
I pushed a change bringing this back to local irods for now as we currently don't have a way to cleanup zips on federated irods.  We should discuss this again once @alvacouch and @hyi are back in the office.